modules/battery_manager/php/testoptionscontroller.class.inc

```diff
@@ -0,0 +1,127 @@
+{
+    /**
+     * This acts as an Ajax enpoint that handles all action requests from the
+     * Battery Manager Module.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $db     = \Database::singleton();
+        $user   = $request->getAttribute('user');
+        $method = $request->getMethod();
+
+        switch($method) {
+        case 'GET':
+            return (new \LORIS\Http\Response())
+                ->withHeader('Content-Type', 'application/json')
+                ->withStatus(200)
+                ->withBody(new \LORIS\Http\StringStream($this->getOptions()));
+        }
+
+        return (new \LORIS\Http\Response())
+            ->withHeader('Content-Type', 'application/json')
+            ->withStatus(400)
+            ->withBody(new \LORIS\Http\StringStream('Unspecified Request'));
+    }
+
+    /**
+     * Gets the field options for this module.
+     *
+     * @return \LORIS\Data\Provisioner
+     */
+    function getOptions()
+    {
+        $options = [
+                    'instruments' => \Utility::getAllInstruments(),
+                    'stages'      => $this->getStageList(),
+                    'subprojects' => \Utility::getSubprojectList(null),
+                    'visits'      => \Utility::getVisitList(),
+                    'sites'       => \Utility::getSiteList(false),
+                    'firstVisits' => $this->getYesNoList(),
+                    'active'      => $this->getYesNoList(),
+                   ];
+        return json_encode($options);
+    }
+
+    /**
+     * Return associative array of stages.
+     *
+     * @return array
+     */
+    function getStageList() : array
+    {
+        return array(
+                "Not Started"   => 'Not Started',
+                "Screening"     => 'Screening',
+                "Visit"         => 'Visit',
+                "Approval"      => "Approval",
+                "Subject"       => "Subject",
+                "Recycling Bin" => "Recycling Bin",
+               );
+    }
+
+    /**
+     * Return associative array of yes and no values.
+     *
+     * @return array
+     */
+    function getYesNoList() : array
+    {
+        return array(
+                'Y' => 'Yes',
+                'N' => 'No',
+               );
+    }
+
+    /**
+     * Returns a response to the front end.
+     *
+     * @param int    $status of response
+     * @param string $body   of response
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    function response(int $status, string $body) : ResponseInterface
+    {
+        return (new \LORIS\Http\Response())
+            ->withHeader('Content-Type', 'application/json')
+            ->withStatus($status)
+            ->withBody(new \LORIS\Http\StringStream($body));
+    }
+}
```

Comment:
I'm wondering if we could have a library function like this that we could place somewhere.

Reply:
I am using JsonResponse objects in my development
https://github.com/aces/Loris/pull/4244/files#diff-b8deb5cd55aeb6e5b83f8c1955992057R94